src/display/canvas.js

```diff
@@ -1089,6 +1089,9 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
 
       if (isPatternFill) {
         ctx.save();
+        var m = this.baseTransform ||
+                this.ctx.mozCurrentTransform.slice();
+        ctx.setTransform(m[0], m[1], m[2], m[3], m[4], m[5]);
```

Comment:
> So this code just performs: if (this.baseTransform) { ctx.setTransform.apply(ctx, this.baseTransform); }? (Now I wonder if we ever will get this.baseTransform == null.)
I initially just used `this.baseTransform`, got `null` in some test cases from the test suite, and saw that in other places the expression `this.baseTransform || this.ctx.mozCurrentTransform.slice();` was used. I did not know that the current tranformation was actually stored in `this.ctx.mozCurrentTransform`, (although now it makes a lot of sense).
You are right that this piece of code could better be `if (this.baseTransform) { ctx.setTransform.apply(ctx, this.baseTransform); }`.

> Shall stroke have the same statements?
I some manner of speaking: "Yes". However, when transforming the stroke, the stroke width would be transformed as well. I do not know if other side effects occur when using `ctx.setTransform()`, that's why I initially thought that it may be a better idea to not use `ctx.save()` and `ctx.restore()`, but to alter the matrices of the patterns directly.